fabric-events-lifecycle-v0/src/main/java/net/fabricmc/fabric/api/event/entity/EntityTickCallback.java

```diff
@@ -0,0 +1,16 @@
+package net.fabricmc.fabric.api.event.entity;
+
+import net.fabricmc.fabric.impl.event.EntityTypeCaller;
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.EntityType;
+
+/**
+ * Callback for ticking an entity.
+ */
+public interface EntityTickCallback<T extends Entity> {
+	public static <T extends Entity> EntityTickCallback<T> event(EntityType<T> type) {
+		return ((EntityTypeCaller)type).getEntityEvent();
+	}
+
+	void tick(Entity entity);
```

Comment:
that feeling when you have generics yet completely omit them in the one place where the event caller could use them

Reply:
oh crap, I entirely forgot to fix that lol

Reply:
Now to figure out how to make it stop screaming at me that `tick(capture<?>) in EntityTickCallback cannot be applied to (net.minecraft.entity.Entity)`

Reply:
Cast; also, you want EntityTypeCaller to be non-generic if you're not gonna use it generically anyway

Reply:
Tried casting through `(Entity)(Object)this` but it doesn't seem to be working.

Reply:
Cast the EntityTickCallback, not the Entity...

Reply:
ahhhh yeah that'd make more sense